### PR TITLE
add local license checks

### PR DESCRIPTION
**Pull request by valentinsulzer**
- function to save the config to the user config dir from platformdirs
- functions to load the config and check the license

### CODE REVIEW

File: ionwizard/validate.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import platformdirs
+import yaml
+from pathlib import Path
+import requests
+from .library_wizard import KEYGEN_ACCOUNT_ID
```
[the remark this answers is not in the document]
Comment (kratman) [reply ionwizard/validate.py:5]: Actually I fixed this a different way in my review PR

File: ionwizard/validate.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        with open(config_file, "r") as f:
+            config = yaml.safe_load(f)
+        libraries = config.get("ionworks", {}).get("libraries", {})
+        for library in libraries:
+            if library["library"] == library_name:
+                return library["key"]
+    return None
+
+
+def library(library_name, custom_library_key=None):
```
[the remark this answers is not in the document]
Comment (kratman) [reply ionwizard/validate.py:23]: PyCharm flags the scope shadowing in the for loop
image

File: ionwizard/library_wizard.py [hunk 3]
```diff
@@ -20,26 +25,46 @@ def install_library(lib_name, web_address):
...
+        with open(config_path, "w") as f:
+            yaml.dump({"ionworks": config}, f)
 
 
 def run():
     try:
         config_file = sys.argv[1]
-        IonWorksPipWizard.install_from(IonWorksPipWizard.process_config(config_file))
+        processed_config = IonWorksPipWizard.process_config(config_file)
+        # IonWorksPipWizard.install_from(processed_config)
```
Comment (kratman) [inline ionwizard/library_wizard.py:66]: I think if we want a script to just save the config, then we can make one for that, but this should still install packages if needed.

**Review by kratman — CHANGES_REQUESTED** [review]
I put the changes up in another branch

File: ionwizard/validate.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (valentinsulzer) [reply ionwizard/validate.py]: the idea was to call it as `ionwizard.validate.library` (in which case the action is "validate library") but I'm not too fussed about this

**Pull request merged**